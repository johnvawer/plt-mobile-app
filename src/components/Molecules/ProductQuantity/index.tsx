import React from 'react'
import { View, Text, StyleSheet } from 'react-native'

import Button from '../../Atoms/Button'

interface Props {
  onPressMore: () => void
  onPressLess: () => void
  quantity: number
}

const ProductQuantity = ({
  onPressMore,
  onPressLess,
  quantity
}: Props): JSX.Element => (
  <View style={styles.container}>
    <Button buttonText='-' onPress={onPressLess} />
    <Text>{quantity}</Text>
    <Button buttonText='+' onPress={onPressMore} />
  </View>
)

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center'
  }
})

export default ProductQuantity
