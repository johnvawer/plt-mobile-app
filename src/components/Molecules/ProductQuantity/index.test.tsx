import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import ProductQuantity from '.'

describe('<ProductQuantity />', () => {
  let props: {
    onPressMore: () => void
    onPressLess: () => void
    quantity: number
  }
  let onPressMore: jest.Mock
  let onPressLess: jest.Mock

  beforeEach(() => {
    onPressLess = jest.fn()
    onPressMore = jest.fn()
    props = {
      onPressMore,
      onPressLess,
      quantity: 1
    }
  })

  describe('<ProductQuantity />', () => {
    it('render based on the props correctly', () => {
      const { toJSON } = render(<ProductQuantity {...props} />)

      expect(toJSON()).toMatchSnapshot()
    })

    it('should call the onPressMore prop on pressing the more button', () => {
      const { getByText } = render(<ProductQuantity {...props} />)
      const button = getByText('+')
      fireEvent.press(button)

      expect(onPressMore).toHaveBeenCalledTimes(1)
    })

    it('should call the onPressLess prop on pressing the less button', () => {
      const { getByText } = render(<ProductQuantity {...props} />)
      const button = getByText('-')
      fireEvent.press(button)

      expect(onPressLess).toHaveBeenCalledTimes(1)
    })
  })
})
