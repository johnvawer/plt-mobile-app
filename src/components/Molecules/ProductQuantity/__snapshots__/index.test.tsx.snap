// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductQuantity /> <ProductQuantity /> render based on the props correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "black",
        "borderWidth": 1,
        "margin": 5,
        "padding": 10,
      }
    }
  >
    <Text>
      -
    </Text>
  </View>
  <Text>
    1
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "black",
        "borderWidth": 1,
        "margin": 5,
        "padding": 10,
      }
    }
  >
    <Text>
      +
    </Text>
  </View>
</View>
`;
